woocommerce/woocommerce-ios
# [Mobile Payments] Use system plugins, not site plugins response, to drive onboarding (for shop manager's sake)

Closes #4960

Why are we doing this?
- To allow Shop Managers to have a proper on-boarding experience too. This fixes a problem where relying on WordPress's REST API endpoint for plugin enumeration fails for non Administrators by replacing it with a Shop Manager friendly endpoint in WooCommerce core.

What changed?
- Switch from SitePluginStore -> SystemStatusStore in CardPresentPaymentsOnboardingUseCase and tests
- Adds insertSampleSystemPlugin to MockStorageManager+Samples
- The "System Plugins" remote and mapper are actually a "System Status" remote and mapper, so I renamed them SystemPluginsRemote, SystemPluginsMapper -> SystemStatusRemote, SystemStatusMapper to make maintenance and understanding simpler down the road (and since we need more than just the active plugins nested object now
- Updated the JSON sample with a new one that includes active and inactive plugins
- Changed SystemStatusMapper to return both active and inactive plugins in a single array, using the networkActivated as a "hack" for now to indicate active or inactive state. See #5269 for our next step to undo this "hack"
- Updated CardPresentPaymentsOnboardingUseCase to look at networkActivated to determine if WooCommerce Payments is active or not
- Merged up from `release/7.8` to pick up fix for #5262 

What's next?
- Undo the networkActivated workaround we've introduced here by doing this with Core Data properly in #5269

To Test:

```
Install WooCommerce Payments 3.1.0 on the site but do not activate it
Log in to the app as a user with Shop Manager role
Navigate to Settings (gear) and then In-Person Payments
Tap on In-Person Payments
The “Connecting to your account” page will appear briefly
Ensure you are prompted to “Activate WooCommerce Payments”
Activate WooCommerce Payments in wp-admin
Tap on “Refresh After Activating”
The “Connecting to your account” page will appear briefly
Ensure you are now able to Order, Manage, or read the Manual for the card reader
Exit the App

Install WooCommerce Payments 2.4.0 on the site but do not activate it
Log in to the app as a user with Shop Manager role
Navigate to Settings (gear) and then In-Person Payments
Tap on In-Person Payments
The “Connecting to your account” page will appear briefly
Ensure you are prompted that you have installed an “Unsupported WooCommerce Payments version”
Activate WooCommerce Payments 2.4.0
Tap on “Refresh After Updating”
The “Connecting to your account” page will appear briefly
Ensure you are STILL prompted that you are running an “Unsupported WooCommerce Payments version”
In wp-admin, update to the latest WooCommerce Payments (e.g. 3.1.0)
Tap on “Refresh After Updating”
The “Connecting to your account” page will appear briefly
Ensure you are now able to Order, Manage, or read the Manual for the card reader

Activate all plugins 
Ensure settings still loads ( since `"inactive_plugins":[]`)

```

Props @fernandofgfer for the initial implementation of the system status active plugins remote and mapper

Update release notes:
